eventsource-client/src/client.rs

```diff
@@ -63,22 +65,32 @@ pub struct ClientBuilder {
     headers: HeaderMap,
     reconnect_opts: ReconnectOptions,
     read_timeout: Option<Duration>,
+    last_event_id: String,
 }
 
 impl ClientBuilder {
     /// Create a builder for a given URL.
     pub fn for_url(url: &str) -> Result<ClientBuilder> {
         let url = url
             .parse()
-            .map_err(|e| Error::InvalidParameter(Box::new(e)))?;
+            .map_err(|_| Error::HttpRequest(StatusCode::BAD_REQUEST))?;
```

Comment:
Why is this bad request now? Seems like an odd error for the client builder to return.

Reply:
I must've transcribed this on auto-pilot.

Reply:
Ah, it's because HttpRequest error can only hold status codes now.